library/CMakeLists.txt

```diff
@@ -192,7 +192,20 @@ if(CMAKE_COMPILER_IS_GNUCC)
 endif(CMAKE_COMPILER_IS_GNUCC)
 
 if(CMAKE_COMPILER_IS_CLANG)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-declarations -Wmissing-prototypes -Wdocumentation -Wno-documentation-deprecated-sync -Wunreachable-code")
+    execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpversion OUTPUT_VARIABLE CLANG_VERSION)
+
+    # Clang version 15 or above introduces new rules for -WDocumentation that
+    # state that we cannot have a non-empty description after \return, \retval
+    # VALUE or param NAME. However, the Doxygen documentation says that the
+    # description runs to the end of the paragraph and doesn't say that it can't
+    # be empty, and the Doxygen tool is happy with our code. With
```

Comment:
This is all a bit too much information here — the point is that our code has empty descriptions in lots of places and recent clang is the only tool that complains.